SillySCP/Handlers/AntiSurface.cs

```diff
@@ -1,117 +1,98 @@
+    private CoroutineHandle _handle;
 
-    private IEnumerator<float> SurfaceChecker()
+    private static IEnumerator<float> SurfaceChecker()
     {
-        yield return Timing.WaitForSeconds(120);
-        
-        if (Exiled.API.Features.Player.List.Count(player => player.IsAlive && !player.IsScp) > 3) 
-            yield break;
+        HashSet<Exiled.API.Features.Player> lastKnownPlayers = new (GetSurfacePlayers());
+        DateTime firstCalled = DateTime.Now;
+        bool isFirstRun = true;
 
-        List<Exiled.API.Features.Player> surfacePlayers = Room.Get(RoomType.Surface).Players.ToList();
-        
-        if (surfacePlayers.Count == 0 || 
-            surfacePlayers.Count(p => p.IsScp) > 1 || 
-            surfacePlayers.Select(p => p.Role.Team).Distinct().Count() > 1)
-            yield break;
-
-        Team team = surfacePlayers[0].Role.Team;
-        int playerCount = surfacePlayers.Count;
-        
-        Door gateA = Door.Get(DoorType.ElevatorGateA);
-        Door gateB = Door.Get(DoorType.ElevatorGateB);
-        Door ezA = Door.Get(DoorType.GateA);
-        Door ezB = Door.Get(DoorType.GateB);
-        
-        int closerToACount = surfacePlayers.Count(player => 
-            Vector3.Distance(player.Position, gateA.Position) <= Vector3.Distance(player.Position, gateB.Position));
-        
-        ezA.IsOpen = ezB.IsOpen = true;
-        ezA.Lock(DoorLockType.Warhead);
-        ezB.Lock(DoorLockType.Warhead);
-        
-        string message;
-        string translation;
-        
-        if (playerCount != closerToACount)
-        {
-            message = $"{playerCount} {Spaceify(team.ToString())} personnel located on surface";
-            translation = $"{playerCount} {Spaceify(team.ToString())} personnel located on Surface";
-        }
-        else
+        foreach (int seconds in Seconds)
         {
-            string nearGate = closerToACount > 0 ? "a" : "b";
-            string nearGateTranslation = closerToACount > 0 ? "A" : "B";
-            message = $"{playerCount} {Spaceify(team.ToString())} personnel located near gate {nearGate}";
-            translation = $"{playerCount} {Spaceify(team.ToString())} personnel located near Gate {nearGateTranslation}";
+            if (TotalSeconds(_wave) < seconds) 
+                continue;
+    
+            yield return Timing.WaitUntilTrue(() => TotalSeconds(_wave) <= seconds);
+    
+            if (isFirstRun && (DateTime.Now - firstCalled).TotalSeconds < 29)
+            {
+                lastKnownPlayers = new (GetSurfacePlayers());
+                isFirstRun = false;
+                continue;
+            }
+    
+            IEnumerable<Exiled.API.Features.Player> currentPlayers = GetSurfacePlayers();
+    
+            foreach (Exiled.API.Features.Player player in currentPlayers)
+            {
+                if (lastKnownPlayers.Contains(player))
+                {
+                    player.Kick("Holding up the round on surface.");
```

Comment:
8.3.4. Every automated kick or ban (except ones issued by native game functions) must
be clearly described to the kicked/banned Player as an automated action not
related to native game function, unless it clearly does not look like an automated
action.
```suggestion
                    player.Kick("This is an automated kick performed by a plugin:\n Holding up the round on surface.");
```